### PR TITLE
8244075: Accelerator of ContextMenu's MenuItem is not removed when ContextMenu is removed from Scene

### PR DESCRIPTION
**Pull request by arapte**
Issue:
There are several issues related to Accelerator of MenuItem of a ConextMenu set on Control.
1. Accelerator of ContextMenu's MenuItem is not removed when ContextMenu is removed from Scene
2. Accelerator is not updated correctly when the Control is removed from a Scene and Added to a different Scene
3. Accelerator is not removed from Scene when the anchor node is removed from Scene and then it's ContextMenu is set to null

Fix:
The accelerator should be added or removed correctly according to the Scene property of the anchor node. 
The issue#3 in above list is only fixed for Button and fails for other Controls. A test is added for this scenario and I shall report a new issue to address it.
Added tests that fails before and pass after the fix.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8244075](https://bugs.openjdk.java.net/browse/JDK-8244075): Accelerator of ContextMenu's MenuItem is not removed when ContextMenu is removed from Scene

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - **Reviewer**)
 * [Ajit Ghaisas](https://openjdk.java.net/census#aghaisas) (@aghaisas - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jfx pull/515/head:pull/515` \
`$ git checkout pull/515`

Update a local copy of the PR: \
`$ git checkout pull/515` \
`$ git pull https://git.openjdk.java.net/jfx pull/515/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 515`

View PR using the GUI difftool: \
`$ git pr show -t 515`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jfx/pull/515.diff">https://git.openjdk.java.net/jfx/pull/515.diff</a>

</details>

### CODE REVIEW

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/AcceleratorParameterizedTest.java [hunk 1]
```diff
@@ -297,4 +300,77 @@ private void assertSceneDoesNotContainKeyCombination(KeyCombination keyCombinati
...
+        Scene diffScene = new Scene(root); // Add the children to a different scene
+        sl.getStage().setScene(diffScene);
+        kb = new KeyEventFirer(item1, diffScene);
+
+        assertEquals(1, getListenerCount(item1.acceleratorProperty()));
+        kb.doKeyPress(KeyCode.DIGIT1, KeyModifier.ALT);
+        assertEquals(2, eventCounter);
+    }
+
+    @Ignore("Passes only for Button")
```
Comment (kevinrushforth) [inline modules/javafx.controls/src/test/java/test/javafx/scene/control/AcceleratorParameterizedTest.java:352]: Can you file the follow-on bug and then update this comment to use that bug ID?

**Review by kevinrushforth — APPROVED** [review]
Looks good. I left one comment on the test. When you update it to include the bug ID of the follow-on bug, I'll re-review it.

File: modules/javafx.controls/src/test/java/test/javafx/scene/control/AcceleratorParameterizedTest.java (hunk 1 above)
Comment (arapte) [reply modules/javafx.controls/src/test/java/test/javafx/scene/control/AcceleratorParameterizedTest.java]: Updated with bug ID [JDK-8268374](https://bugs.openjdk.java.net/browse/JDK-8268374)

**Review by kevinrushforth — APPROVED** [review]

**Review by aghaisas — APPROVED** [review]

**Pull request closed without merging**